deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+
+  // Generates scripts necessary for deployment of each network
+  generateScripts(protocol, network, template) {
+    let scripts = [];
+
+    let location = this.getLocation(protocol, network);
+
+    scripts.push("rm -rf build");
+    scripts.push("rm -rf generated");
+    scripts.push("rm -rf results.txt");
+    scripts.push("rm -rf configurations/configure.ts");
+    scripts.push("rm -rf subgraph.yaml");
+
+    scripts.push(
+      "npm run prepare:yaml --PROTOCOL=" +
+        protocol +
+        " --NETWORK=" +
+        network +
+        " --TEMPLATE=" +
+        template
+    );
+    if (
+      this.data[protocol]["networks"][network]["options"][
+        "prepare:constants"
+      ] == true
+    ) {
+      scripts.push(
+        "npm run prepare:constants --PROTOCOL=" +
+          protocol +
+          " --NETWORK=" +
+          network
+      );
+    }
+    scripts.push("graph codegen");
+
+    // We don't want to deploy if we are building or just testing.
+    if (this.type == "deploy" || this.type == "") {
+      scripts.push(this.getDeploymentScript(location));
+    } else {
+      scripts.push("graph build");
+    }
+
+    this.scripts.set(location, scripts);
+  }
+
+  // Just flattens the first level of the json file so you have networks as keys, instead of protocol type.
+  flattenFirstJsonLevel() {
+    let result = {};
+    Object.keys(this.data).forEach((key1) =>
+      Object.keys(this.data[key1]).forEach(
+        (key2) => (result[key2] = this.data[key1][key2])
+      )
+    );
+    this.data = result;
+  }
+
+  // Makes sure the input arguments for the deployments you want are sensible.
+  checkValidScope() {
+    if (!this.protocol && !this.fork) {
+      throw "--PROTOCOL, --NETWORK, --FORK: Please specify a protocol, protocol and network, or fork.";
+    } else if (this.fork && (this.protocol || this.network)) {
+      throw "--PROTOCOL, --NETWORK, --FORK: If you specify a fork, you must not specify a protocol or network";
+    }
+  }
+
+  // Checks if the input arguments are valid for the deployment location.
+  checkValidService() {
+    switch (this.service) {
+      case "subgraph-studio":
+      case "studio":
+      case "s":
+      case "decentralized-network":
+      case "d":
+        return true;
+      case "hosted-service":
+      case "hosted":
+      case "h":
+        return true;
+      case "cronos-portal":
+      case "cronos":
+      case "c":
+        return true;
+      default:
+        throw (
+          "--SERVICE: Service is not valid or is missing: service=" +
+          this.service
+        );
+    }
+  }
+
+  // Makes sure the version specified in the json file is valid.
+  checkValidVersion(version) {
+    this.checkVersionLengthIsTwo(version);
+    this.checkVersionIsNumber(version);
+  }
+
+  checkVersionLengthIsTwo(version) {
+    if (version.split(".").length - 1 != 2) {
+      throw "See deployment.json: (1) version is not valid - must be in format x.x.x (e.g. 1.3.1)";
+    }
+  }
```

Comment:
XD. Haha, yeah... that makes sense. Although, this is a point I wanted to discuss. Since we may not be including the version in the deployment.json, I might scrap the version checks.